src/Rule/UnusedVariable.php

```diff
@@ -64,9 +68,12 @@ public function check(TokenStream $tokens)
                         ]);
                         break;
                     case 'if':
-                    case 'for':
                         $this->skip($tokens, 3);
                         break;
+                    case 'for':
+                        $scope->declare($tokens->look(4)->getValue(), $tokens->look(4));
```

Comment:
does this work for the case where the for loop iterates with both key and value ?